File: schemaorg/gameserverstatus.go
package schemaorg

import "encoding/json"

// GameServerStatus see : https://schema.org/GameServerStatus
type GameServerStatus struct {

typeContext

Enumeration

// SupersededBy see : https://schema.orghttp://meta.schema.org/supersededBy
// Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
SupersededBy interface{} `json:"supersededBy"` // types : Class Enumeration Property

}

func (v *GameServerStatus) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "GameServerStatus"

	return json.Marshal(v)
}

File: schemaorgphp/classes/game_server_status.php
<?php

class GameServerStatus extends Enumeration implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'GameServerStatus';
	
	/**
	 * Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
	 * see : http://meta.schema.org/supersededBy
	 * @var \Class|\Class[]|\Enumeration|\Enumeration[]|\Property|\Property[]
	 */
	public var $superseded_by;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'GameServerStatus'
		);
		
		$serialized = so_json_serialize( $this->superseded_by );
		if ( ! empty( $serialized ) ) {
			$out['supersededBy'] = $serialized;
		}
		
		return $out;
	}
}
